db/migrate/20180403144853_convert_vm_attrs_to_hash.rb

```diff
@@ -0,0 +1,45 @@
+class ConvertVmAttrsToHash < ActiveRecord::Migration[5.1]
+  def up
+    transform ActionController::Parameters, :to_h
+  end
+
+  def down
+    transform ActiveSupport::HashWithIndifferentAccess, :to_params
+  end
+
+  def transform(from, transform_method)
+    ComputeAttributeHack.unscoped.all.in_batches do |batch|
+      batch.each do |comp_attr|
+        attributes = comp_attr.vm_attrs
+        if YAML.load(attributes).is_a? from
+          comp_attr.vm_attrs = send transform_method, attributes
+          comp_attr.save!
+        end
+      end
+    end
+  end
+
+  def to_h(attr)
+    attr.gsub(yml_params_hash, yml_hash).gsub(yml_params_obj, yml_hash)
```

Comment:
`sub` might be enough, especially on large strings, it would be faster

Reply:
You can use `gsub!` to reduce the object allocations and data copying here (`gsub` creates a new string with the result, `gsub!` makes the change in place)